plugin/src/main/groovy/grails/plugin/springsecurity/ui/strategy/DefaultRegistrationCodeStrategy.groovy

```diff
@@ -40,6 +41,14 @@ class DefaultRegistrationCodeStrategy implements RegistrationCodeStrategy {
 		springSecurityUiService.register user, password, salt
```

Comment:
I've tested this locally and, by default, if `requireForgotPassEmailValidation = false`, newly registered users are created with a locked account.

Was this intentional?

Reply:
I will do some testing later one but it seems that this parameter should not effect if the newly created accounts are set to be locked out.  The configuration of requireEmailValidation should have some effect on this because it looks like when new accounts are created they are supposed to be locked out until the e-mail is clicked.  If this was set to false then I would think it would be unlocked sine the token step happens behind the scene.

Reply:
I've tested this again and there appears to be a bug.

Yes, the token verification step does set `User.accountLocked = false`, but at some point after that it is persisted to the database as `User.accountLocked = true`

To duplicate this bug, create a simple grails app which utilizes your plugin changes, set `requireForgotPassEmailValidation = false` and register a new user.

Once registered, login as an admin and user the `UserController` to search for the new user and view their details.  You will see that their account is locked.

Reply:
@amondel2 I found the cause of the issue.  

The `User.accountLocked` field will be true if the default role name being associated with the user doesn't exist.

This is not a bug, but the code should be improved to avoid this.

I'll writeup a separate issue to tackle this improvement since it is unrelated to your changes.

Thanks!

Reply:
Thank you for doing the RCA!